UI/EncyclopediaDetailPanel.cpp

```diff
@@ -2287,8 +2287,29 @@ namespace {
         for (SpeciesManager::iterator it = species_manager.begin();
                 it != species_manager.end(); ++it)
         {
-            if (it->second && (it->second->Tags().find("CTRL_ALWAYS_REPORT") != it->second->Tags().end()))
-                    species_names.insert(it->first);
+            if (it->second) {
+                const std::string& species_str = it->first;
+                const std::set<std::string>& species_tags = it->second->Tags();
+                if (species_tags.find("CTRL_ALWAYS_REPORT") != species_tags.end()) {
```

Comment:
Could you put some comments amongst this code to explain what you're checking / doing? it's a bit dense to read through and interpret otherwise...